doc/source/whatsnew/v0.22.0.txt

```diff
@@ -129,6 +124,10 @@ Other API Changes
 - :func:`DataFrame.from_items` provides a more informative error message when passed scalar values (:issue:`17312`)
 - When created with duplicate labels, ``MultiIndex`` now raises a ``ValueError``. (:issue:`17464`)
 - Building from source now explicity requires ``setuptools`` in ``setup.py`` (:issue:`18113`)
+- :func:`Series.fillna` now raises a ``TypeError`` instead of a ``ValueError`` when passed a list, tuple or DataFrame as a ``value`` (:issue:`18293`)
```

Comment:
looks like you picked up some other commits here.

Reply:
those are merged on master now, so I think merging them as is won't harm (or duplicate).

Reply:
I am not sure if it will do harm or not, but can you still fix this? (it makes reviewing harder as there are unrelated changes) 
In principle

```
git fetch upstream
git merge upstream/master
```

should do it

Reply:
Merged